sdk/core/src/services/UserManagementService.ts

```diff
@@ -26,7 +26,8 @@ import { extractSDKVersion } from '../_helpers'
 const generatePassword = async () => {
   const randomPassword = (await randomBytes(32)).toString('hex')
   // Make first found letter uppercase because hex string doesn't meet password requirements
-  return randomPassword.replace(/[a-f]/, 'A')
+  // Special characters at the end of password to make comply with cognito requirements
+  return randomPassword.replace(/[a-f]/, 'A') + '!'
```

Comment:
We definitely should write some TODO here...
```
// TODO: Actually generate a random password instead of simply replacing first letter of hexadecimal password with 'A' and appending an exclamation mark
```

Reply:
Good catch. Sure, let's keep it for later